packages/admin/cms-admin/src/dam/MoveDamItemDialog/ChooseFolder.tsx

```diff
@@ -0,0 +1,312 @@
+                                            });
+                                        }
+                                    }}
+                                    message={folderMatches ? <MarkedMatches text={folder.name} matches={folderMatches} /> : folder.name}
+                                    offset={20 + 36 * level}
+                                    isChosen={selectedId === folder.id}
+                                    onClick={() => {
+                                        onFolderClick(folder.id);
+                                    }}
+                                    style={style}
+                                />
+                            );
+                        }}
+                    </List>
+                );
+            }}
+        </AutoSizer>
+    );
+};
+
+const StyledListItem = styled(ListItem)<{ offset: number; isChosen?: boolean }>`
+    display: flex;
+    justify-content: space-between;
+
+    height: 56px;
+    background: white;
+    cursor: pointer;
+    padding-left: ${({ offset }) => offset}px;
+
+    border-left: solid 2px transparent;
+    border-right: solid 2px transparent;
+    border-bottom: solid 1px ${({ theme }) => theme.palette.grey[100]};
+
+    &:last-child {
+        border-bottom: none;
+    }
+
+    &:hover {
+        border-left: solid 2px ${({ theme }) => theme.palette.primary.main};
+        border-right: solid 2px ${({ theme }) => theme.palette.primary.main};
+        background: ${({ theme }) => theme.palette.grey[50]};
+    }
+
+    ${({ isChosen, theme }) => {
+        return isChosen
+            ? `
+                border-top: solid 2px ${theme.palette.primary.main};
+                border-bottom: solid 2px ${theme.palette.primary.main};
+                border-left: solid 2px ${theme.palette.primary.main};
+                border-right: solid 2px ${theme.palette.primary.main};
+                
+                &:last-child {
+                    border-bottom: solid 2px ${theme.palette.primary.main};
+                }
+            `
+            : null;
+    }}
+`;
+
+interface ChooseFolderItemProps {
+    Icon?: React.ComponentType<SvgIconProps>;
+    onIconClick?: () => void;
+    onClick?: () => void;
+    message: React.ReactNode;
+    offset: number;
+    isChosen?: boolean;
+    style: React.CSSProperties;
+}
+const ChooseFolderItem = ({ Icon, onIconClick, onClick, message, offset, isChosen = false, style }: ChooseFolderItemProps) => {
+    return (
+        <StyledListItem style={style} offset={Icon ? offset : offset + 20 + 16} isChosen={isChosen} onClick={onClick}>
+            <div>
+                {Icon && (
+                    <Icon
+                        sx={{ marginRight: "20px" }}
+                        onClick={(event) => {
+                            event.stopPropagation();
+                            onIconClick?.();
+                        }}
+                    />
+                )}{" "}
```

Comment:
The space should only be rendered if an icon exists. Or is this even necessary, as you're adding a right margin anyway?